Sources/Swift/Protocol/SentryAttributeValuable.swift

```diff
@@ -0,0 +1,145 @@
+public protocol SentryAttributeValuable {
+    var asAttributeValue: SentryAttributeValue { get }
+}
+
+extension String: SentryAttributeValuable {
+    public var asAttributeValue: SentryAttributeValue {
+        return .string(self)
+    }
+}
+
+extension Bool: SentryAttributeValuable {
+    public var asAttributeValue: SentryAttributeValue {
+        return .boolean(self)
+    }
+}
+
+extension Int: SentryAttributeValuable {
+    public var asAttributeValue: SentryAttributeValue {
+        return .integer(self)
+    }
+}
+
+extension Double: SentryAttributeValuable {
+    public var asAttributeValue: SentryAttributeValue {
+        return .double(self)
+    }
+}
+
+extension Float: SentryAttributeValuable {
+    public var asAttributeValue: SentryAttributeValue {
+        return .double(Double(self))
+    }
+}
+
+extension Array: SentryAttributeValuable {
+    public var asAttributeValue: SentryAttributeValue {
+        // Arrays can be heterogenous, therefore we must validate if all elements are of the same type.
+        // We must assert the element type too, because due to Objective-C bridging we noticed invalid conversions
+        // of empty String Arrays to Bool Arrays.
+        if Element.self == Bool.self, let values = self as? [Bool] {
+            return .booleanArray(values)
+        }
+        if Element.self == Double.self, let values = self as? [Double] {
+            return .doubleArray(values)
+        }
+        if Element.self == Float.self, let values = self as? [Float] {
+            return .doubleArray(values.map(Double.init))
+        }
+        if Element.self == Int.self, let values = self as? [Int] {
+            return .integerArray(values)
+        }
+        if Element.self == String.self, let values = self as? [String] {
+            return .stringArray(values)
+        }
+        if let values = self as? [SentryAttributeValuable] {
+            return castArrayToAttributeValue(values: values)
+        }
+        return .stringArray(self.map { element in
+            String(describing: element)
+        })
+    }
+
+    private func castArrayToAttributeValue(values: [SentryAttributeValuable]) -> SentryAttributeValue {
```

Comment:
`m`: Honestly, I find that a bit confusing. I can do this, and the compiler doesn't complain. When looking at this code, I could assume that we support arrays with different types for the attributes, but instead, we silently convert them to a string array. Instead, it would be great if the following code wouldn't even compile.
```
let attribute : [SentryAttributeValuable] = [1,2,3, 3.3, "test"]
let attributeValue = attribute.asAttributeValue
```

Reply:
OK, maybe we need a completely different approach if it doesn't work with the current one. I can try to come up with something, but that will take some time, and that will have to wait after the holidays.